ci/Dockerfile

```diff
@@ -26,14 +26,21 @@ ENV PATH /root/.cargo/bin/:$PATH
 # add required rustup components
 RUN rustup component add rustfmt llvm-tools-preview clippy
 
+ENV CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
+
 # install build tools
-RUN cargo install cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-sort cargo-make cargo-cache sccache \
+RUN cargo install cargo-quickinstall --locked
+RUN cargo quickinstall cargo-binstall
+RUN cargo binstall -y --no-symlinks cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-sort cargo-make cargo-cache \
```

Comment:
Why do we need this trick 🥵

Reply:
What trick do you mean

Reply:
Is there any notable difference between `quickinstall` and `binstall`?

Reply:
Look at quickinstall‘s GitHub release you will know what it is.

binstall will try quickinstall and a crate’s GitHub release

Reply:
So binstall is more powerful. But a drawback is slow resolve https://github.com/cargo-bins/cargo-binstall/issues/776